### PR TITLE
Movie details page

### PR DESCRIPTION
**Pull request by 3xer**
Details sidan + screenings är klar kolla om ni tycker designen ser bra ut eller nå annat som kan kanske ändras

### CODE REVIEW

File: src/app/movies/[id]/page.tsx [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
 import Header from "@/components/Header";
+import MovieDetails from "@/components/MovieDetails";
+import Screenings from "@/components/Screenings";
+import { useSearchParams } from "next/navigation";
 
-export default function MoviePage() {
+export default async function DetailsPage({ params }: { params: { id: string }}) {
+  const details = await getMovie(params.id);
+  const screening = await getScreeningById(params.id)
+  //console.log('screenings', screening)
```
Comment (solidtop) [inline src/app/movies/[id]/page.tsx:10]: Glöm inte att ta bort denna :)

File: src/app/page.tsx [hunk 2]
```diff
@@ -6,6 +6,7 @@ import * as api from "@/utils/api";
 export default async function Home() {
   const movies = await api.getMovies();
 
+  const loggedIn: boolean = false;
```
Comment (solidtop) [inline src/app/page.tsx:9]: Vad gör denna "loggedIn" här? Den verkar inte användas, kanske kan tas bort?

File: src/components/Screenings.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { FC } from "react"
+import { Screening } from "@/types"
+import { link } from "fs"
+import Link from "next/link"
+type x = {
+    Screening: Screening[]
+}
```
Comment (solidtop) [inline src/components/Screenings.tsx:7]: Ett förslag är att döpa om type "x" till "ScreeningsProps" för mer tydlighet. Samt variabeln "Screening" kanske kan döpas om till "screenings" för att förmedla att den innehåller flera screenings. Vad tros?

File: src/components/Screenings.tsx
Comment (yvette809) [inline src/components/Screenings.tsx:1]: Since you did not use the imported link from fs I think you should take out that line

File: src/app/movies/[id]/page.tsx (hunk 1 above)
Comment (3xer) [reply src/app/movies/[id]/page.tsx:10]: Inser nu att jag inte kollade igenom allt så noga innan..

File: src/utils/api.ts
Comment (yvette809) [inline src/utils/api.ts:1]: It seems to me that the getSceening() function and the getScreeningById() does the same thing. Can you take out the getSceening() since you did not use it

**Review by yvette809 — COMMENTED** [review]
Good job with the movieDetails page. I think there is need to improve on the styling. You should use a container with margins on both sides and properly align everything. The book button can be more visally appealing by the use of some padding. That is just my suggestion

File: src/utils/api.ts
Comment (solidtop) [reply src/utils/api.ts:1]: getScreening() fetches a single screening with screeningId. getScreeningById() fetches all screenings for a specific movie, with movieId. Maybe the function "getScreeningById" should be renamed to "getScreeningByMovieId()" so it doesn't get confusing 👍🏻

Comment (yvette809) [reply src/utils/api.ts:1]: Ok thanks for the clarification

File: src/app/movies/[id]/page.tsx [hunk 4]
```diff
@@ -1,9 +1,21 @@
...
+  
   return (
     <>
       <Header />
+      <MovieDetails movieDetails={details} />
+      <Screenings Screening={screening} />
     </>
```
Comment (solidtop) [inline src/app/movies/[id]/page.tsx:17]: Lägg gärna till <Footer> komponenten på rad 17 också :)

File: src/components/MovieDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import { FC } from "react";
+import { Movie, Screening } from "@/types";
+import Link from "next/link";
+
+const MovieDetails: FC<Movie> = ({ movieDetails }) => {
```
Comment (solidtop) [inline src/components/MovieDetails.tsx:5]: Får ett typescript varning här att "movieDetails" inte finns på typen Movie. En lätt lösning är att även här göra en MovieDetailsProps type: 
```
type MovieDetailsProps = {
   movieDetails: Movie;
}

const MovieDetails: FC<MovieDetailsProps > = ({ movieDetails }) => {
```

**Review by holst9559 — APPROVED** [review]
Inte den snyggaste designen men allt funkar, oavsett så behöver det mergas in nu för att testa helheten av siten innan deadline.

File: package.json [hunk 6]
```diff
@@ -2,6 +2,7 @@
   "name": "next-kino",
   "version": "0.1.0",
   "private": true,
+
```
Comment (richardolsson) [inline package.json:5]: This is an unnecessary change. Of course it doesn't have any big effect on the application, but it tells me that maybe you are a bit sloppy when deciding what files to include in a commit? I recommend always looking at what files you are committing, and in fact I even recommend looking through the exact lines you're about to commit.

File: src/utils/api.ts [hunk 7]
```diff
@@ -67,21 +68,31 @@ export async function getUserSession(): Promise<User> {
...
 }
 // GET all the screenings
-export async function getScreenings() : Promise<Screening[]> {
+export async function getScreenings(): Promise<Screening[]> {
   const res = await fetch(API_URL + "/screenings/");
   const payload = await res.json();
   return payload;
 }
 // GET screening with specific id
+export async function getScreeningById(
```
Comment (richardolsson) [inline src/utils/api.ts:87]: As discussed in another comment, I would have renamed this to `getScreeningsByMovieId`, i.e. change "screening" to plural and add "movie" to indicate that it's the ID of a movie that's relevant, not the ID of a screening.

File: src/app/movies/[id]/page.tsx [hunk 8]
```diff
@@ -1,9 +1,23 @@
...
+import MovieDetails from "@/components/MovieDetails";
+import Screenings from "@/components/Screenings";
+
+export default async function DetailsPage({
+  params,
+}: {
+  params: { id: string };
+}) {
+  const details = await getMovie(params.id);
+  const screenings = await getScreeningById(params.id);
```
Comment (richardolsson) [inline src/app/movies/[id]/page.tsx:13]: Nice use of async loading in a server-side react component. I still haven't gotten used to this (it didn't use to be possible) so I'm glad you're learning this and don't have to "unlearn" the old way of only retrieving data on the client side using `useEffect()`. 😊

File: src/components/Screenings.tsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+                    </Link>
+                    <div className="w-full h-0.5 col-start-1 col-end-5 my-1 bg-slate-400"></div>
+                    </li>
+                    )
+                })}
+            </ul>
+            
+        )
+    
+
```
Comment (richardolsson) [inline src/components/Screenings.tsx:30]: It feels like you're not being very consistent with indentation here. If you find it difficult to notice this, or even if you do notice it clearly, I recommend using automatic formatting in your editor.

File: src/types.ts [hunk 10]
```diff
@@ -70,6 +78,12 @@ export type SeatObject = {
   seat: number;
   state: string;
 };
+export type movieDetailsProp = {
+  movieDetails: Movie;
+}
+export type ScreeningProps = {
```
Comment (richardolsson) [inline src/types.ts:84]: Types like these that you only use in one places can be declared in those places instead. You don't need to put them in a central module of types and export them if you're only going to use them once.

I would also recommend settling on a naming pattern and sticking to it consistently. Here, I notice that one of the types uses camelCase (lowercase first letter) and the other uses PascalCase (uppercase first letter). Furthermore, one is singular (prop) and the other is plural (props).

**Review by richardolsson — COMMENTED** [review]
Great work! I've left some comments below that hopefully you can pick up something from for the future. 😊
